### ISSUE TITLE
Nested SQL statements do not work anymore

### ISSUE THREAD

**Issue by daseinhorn** [balderdashy/sails#2577]
Hi all

We did a framework update to v0.10.3 and since then, I'm not able to get nested SQL querys to work again. Was there a breaking change in waterline/sails or is this a bug? The statement in the code section has worked just fine before the update from the ~0.10.0-rc9 release of sails.

Code:
```javascript
    User.find().where({
            or: [
                {
                    email: {contains: query}
                },
                {
                    username: {contains: query}
                },
                {
                    and: [
                        {
                            alias: {contains: query}
                        },
                        {
                            useAlias: true
                        }
                    ]
                }
            ]
        }).limit(10)
```

Error:
Sending 500 ("Server Error") response: 
Error (E_UNKNOWN) :: Encountered an unexpected error
: ER_PARSE_ERROR: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '[object Object],[object Object]))) LIMIT 10' at line 1

Stacktrace:
```
at Query.Sequence._packetToError ([project root]node_modules/sails-mysql/node_modules/mysql/lib/protocol/sequences/Sequence.js:48:14)
    at Query.ErrorPacket ([project root]node_modules/sails-mysql/node_modules/mysql/lib/protocol/sequences/Query.js:82:18)
    at Protocol._parsePacket ([project root]node_modules/sails-mysql/node_modules/mysql/lib/protocol/Protocol.js:271:23)
    at Parser.write ([project root]node_modules/sails-mysql/node_modules/mysql/lib/protocol/Parser.js:77:12)
    at Protocol.write ([project root]node_modules/sails-mysql/node_modules/mysql/lib/protocol/Protocol.js:39:16)
    at Socket.<anonymous> ([project root]node_modules/sails-mysql/node_modules/mysql/lib/Connection.js:82:28)
    at Socket.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:765:14)
    at Socket.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:427:10)
    --------------------
    at Protocol._enqueue ([project root]node_modules/sails-mysql/node_modules/mysql/lib/protocol/Protocol.js:135:48)
    at PoolConnection.Connection.query ([project root]node_modules/sails-mysql/node_modules/mysql/lib/Connection.js:185:25)
    at PoolConnection.liveConnection.query ([project root]node_modules/sails-mysql/lib/connections/spawn.js:98:16)
    at PoolConnection.liveConnection.query ([project root]node_modules/sails-mysql/lib/connections/spawn.js:98:16)
    at PoolConnection.liveConnection.query ([project root]node_modules/sails-mysql/lib/connections/spawn.js:98:16)
    at PoolConnection.liveConnection.query ([project root]node_modules/sails-mysql/lib/connections/spawn.js:98:16)
    at PoolConnection.liveConnection.query ([project root]node_modules/sails-mysql/lib/connections/spawn.js:98:16)
    at __FIND__ ([project root]node_modules/sails-mysql/lib/adapter.js:836:20)
    at afterwards ([project root]node_modules/sails-mysql/lib/connections/spawn.js:104:5)
    at [project root]node_modules/sails-mysql/lib/connections/spawn.js:40:7
```

Dependencies:
```json
"dependencies": {
  "sails": "0.10.3",
  "sails-disk": "0.10.6",
  "sails-mysql": "0.10.9",
  "sails-redis": "0.10.3"
 }
```

**Comment by kennysmoothx** [comment]
As a side note I recommend upgrading to `sails-mysql: 0.10.11`

It fixes escaping of periods while using `contains`.

If any of your `contains:` at the moment have a period in them it results in a `ER_PARSE_ERROR`

**Comment by CWyrtzen** [comment]
Is this truly a bug? Is it still present after all upgrading to the most recent versions? Thanks~

**Comment by sailsbot** [comment]
Thanks for posting, @daseinhorn.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by daseinhorn** [comment]
It is still an unsolved issue.
